packages/react-components/react-dialog/src/components/DialogSurface/useDialogSurface.ts

```diff
@@ -59,27 +59,27 @@ export const useDialogSurface_unstable = (
     }
   });
 
+  const backdropSlot = slot(backdrop, {
+    componentType: 'div',
+    required: open && modalType !== 'non-modal',
+    defaultProps: { 'aria-hidden': 'true' },
+    overrides: { onClick: handledBackdropClick },
+  });
+
   return {
-    components: {
-      backdrop: 'div',
-      root: 'div',
-    },
-    backdrop: resolveShorthand(backdrop, {
-      required: open && modalType !== 'non-modal',
-      defaultProps: {
-        'aria-hidden': 'true',
-        onClick: handledBackdropClick,
-      },
-    }),
-    root: getNativeElementProps(as ?? 'div', {
-      tabIndex: -1, // https://github.com/microsoft/fluentui/issues/25150
-      'aria-modal': modalType !== 'non-modal',
-      role: modalType === 'alert' ? 'alertdialog' : 'dialog',
-      'aria-labelledby': props['aria-label'] ? undefined : dialogTitleID,
-      ...props,
-      ...modalAttributes,
-      onKeyDown: handleKeyDown,
-      ref: useMergedRefs(ref, dialogRef),
-    }),
+    backdrop: backdropSlot,
```

Comment:
But honestly, I see no harm in having `overrides` property there, it makes things easier

Reply:
> But honestly, I see no harm in having overrides property there, it makes things easier

How `overrides` makes it easier? :)

```tsx
const slot = slot(props.slot, {
  overrides: {
    onClick: useEventCallback(() => {
      // Hm.. What should I call? 
      if (typeof props.slot === 'object' && props.slot !== null) {
        props.slot.onClick()
      }
    })
  }
})
```

Do we really want to do that?

Reply:
`typeof props.slot === 'object' && props.slot !== null` this is the only problem I see, and it's unnecessary in this scenario, we have `isResolvedShorthand` to avoid that.

Reply:
Correct, but following will be simpler and does not require checks (and we are doing this anyway):

```ts
const foo = slot()

foo.onClick = mergeCallbacks(foo.onClick, () => {})
```